cli/tests/test_git_push.rs

```diff
@@ -1440,36 +1440,57 @@ fn test_git_push_changes_with_name_untracked_or_forgotten(subprocess: bool) {
     // Make sure push still errors if we try to push a bookmark with the same name
     // to a different location.
     let output = work_dir.run_jj(["git", "push", "--named", "b1=@+"]);
```

Comment:
(Aside) I remember being vaguely surprised when I wrote this test case and it failed to push. That seems to be what we want, but my memory of the libgit2 pushing made me think it would have succeeded.

Unfortunately, the feeling wasn't strong enough to double-check that libgit2 is actually being used.

Reply:
FWIW I don’t think we have to ensure parity here. `git.subprocess = false` is in the process of deprecation, so it’s not too surprising if a new feature like `--named` acts subtly differently. (Which is not to say it’s not worth investigating, but I think it’d also be fine to just leave it like this until it’s removed.)

Reply:
(Whereby “leave it like this” I mean after the PR, since it’s weird to run the same test twice and have it not actually test `git2` in the other branch. I just mean it doesn’t seem critical to solve the divergence.)

Reply:
You are probably right. I'm just annoyed feeling a trace of an egg on my face.

Reply:
Perhaps, this is the difference between our ref merge logic and git-native force-with-lease. We'll need to move the branch sideway (instead of descendant) to make both git2 and subprocess fail.

Reply:
Yes, I intentionally picked a borderline case. I guess you're implying it's not worth it to make the behavior consistent in the borderline case? 

I'm surprised this didn't cause a difference between libgit2 and subprocess behavior in some of the detailed tests I once wrote for force-with-lease.

Reply:
> you're implying it's not worth it to make the behavior consistent in the borderline case?

Yes. The main reason afaik is that it's difficult to hook into `git` command to use our conflict resolution logic. The git2 behavior is more consistent with jj, though.

> I'm surprised the underlying cause didn't cause a difference between libgit2 and subprocess behavior in some of the detailed tests I once wrote for force-with-lease.

We have `if subprocess`, etc. in lib tests. ;)

Reply:
Sad. But makes sense, I guess. Fixing it would be messy without git libification, we'd probably have to provide `git` with a way to involve jj as a push negotiation script.